### ISSUE TITLE
Normalization layer supresses ValueError when model is called with bad input shape

### ISSUE THREAD

**Issue by jameshfisher** [tensorflow/tensorflow#47428]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab notebook, and everywhere else tested
- TensorFlow installed from (source or binary):  Colab notebook, and everywhere else tested
- TensorFlow version (use command below): v2.4.1-0-g85c8b2a817f 2.4.1
- Python version: 3.7.10

**Describe the current behavior**
After adding a `tf.keras.layers.experimental.preprocessing.Normalization` layer to a model, badly shaped input is accepted by the model's `.predict`, generating only a warning log line, and returning some nonsense output.

**Describe the expected behavior**
After adding a `tf.keras.layers.experimental.preprocessing.Normalization` layer to a model, badly shaped input should be rejected by the model's `.predict`, raising a `ValueError`.

**Standalone code to reproduce the issue**
[Here is a Colab notebook that shows the issue](https://colab.research.google.com/drive/1ud_9UtZaTwmlL6WTnJkWMXG162DJ57kr?usp=sharing)

**Comment by Saduf2019** [comment]
@jameshfisher 
I ran the code shared in colab but do not face same error reported, please have a look at the [gist here](https://colab.research.google.com/gist/Saduf2019/e61628339732c736514c00a900143689/untitled554.ipynb).

**Comment by jameshfisher** [comment]
@Saduf2019 your gist shows the same problem. See the final code cell in your gist. `model_with_normalization_layer.predict(bad_input)` only produces a `WARNING` log line, and returns a nonsense value of shape `(5,1)`. But what it _should_ do is raise a `ValueError`, and not return any value, because its input has shape `(5,)`, which does not match the shape enforced by its first layer, `tf.keras.Input(shape=(3,))`. This is the normal behavior when a `Normalization` layer is not present (see the line `model_without_normalization_layer.predict(bad_input)` for reference); adding a `Normalization` layer should not change this behavior.

**Comment by Saduf2019** [comment]
@jvishnuvardhan 
I ran the code on nightly as well, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/a80eb59cba101df53317a6b74a8bf35d/untitled555.ipynb).

**Comment by shivaditya-meduri** [comment]
I'd like to work on this.

**Comment by deeb02** [comment]
Thank you for submitting this. It looks like a serious bug. We've raised it to P1.

**Comment by sushreebarsa** [comment]
Was able to reproduce this issue using TF v2.5 ,please check the gist [here](https://colab.research.google.com/gist/sushreebarsa/2c8bd3d92956c4b944be384cf95ba4a9/untitled141.ipynb?authuser=1)..Thanks !

**Comment by sushreebarsa** [comment]
Was able to replicate the issue with TF 2.6.0-dev20210606,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/b3af00c5c5b2dfa4fa6f34668634dba9/untitled168.ipynb?authuser=1) ..Thanks!

**Comment by chunduriv** [comment]
@jameshfisher,
I have tried in Colab with Tensorflow 2.8 and i am seeing `InvalidArgumentError` with `model_with_normalization_layer.predict(bad_input)`. Please, find the gist [here](https://colab.research.google.com/gist/chunduriv/badca4e6aac1b0028109bf628852ecc8/47428.ipynb) and confirm if this issue is resolved for you.Thanks!
